### PR TITLE
fix(select): resolved error when only one option child is set - FE-2547

### PR DESCRIPTION
**Pull request by damianOfKyan**
Fixes FE-2547
Fixes GitHub issue #2567

### Proposed behaviour
Select can accept one or more `<Option />` as children.

### Current behaviour
Select breaks when one `<Option />` is given as child.

### Checklist

- [x] Release notes (Conventional Commits) 
- [x] Unit tests
<dev>- [ ] Cypress automation tests
<dev>- [ ] Storybook added or updated
<dev>- [ ] Theme support
<dev>- [ ] Typescript `d.ts` file added or updated

### Testing instructions
To experience breaking behaviour, on master branch modify the `defaultStory` story in `select.stories.js`.

```diff
const defaultComponent = () => {
   return (
     <State store={ singleSelectStore }>
       <Select ariaLabel='singleSelect' { ...commonKnobs(singleSelectStore) }>
-        { selectOptions }
+        <Option
+          key="a"
+          text="a"
+          value="1"
+        />
       </Select>
     </State>
   );
);
```

browser to:
http://localhost:9001/?path=/story/experimental-select--default
click on select, it should raise the error in the browser console.

Checkout this branch and repeat the steps. There should be no errors.

### CODE REVIEW

File: src/__experimental__/components/select/select.spec.js [hunk 1]
```diff
@@ -462,4 +462,35 @@ describe('Select', () => {
       assertStyleMatch({ color: '#FFFFFF' }, styleWrapper.toJSON(), { modifier: `&:hover ${StyledIcon}` });
     });
   });
+
+  describe('renders select with one option', () => {
+    it('renders correctly', () => {
+      const wrapper = mount(
+        <Select>
```
Comment (mkrds) [inline src/__experimental__/components/select/select.spec.js]: Maybe use `shallow` here because the generated snapshot is 3k LOC

Comment (damianOfKyan) [reply src/__experimental__/components/select/select.spec.js]: fixed.

File: src/__experimental__/components/select/__snapshots__/select.spec.js.snap [hunk 2]
```diff
@@ -99,6 +99,39 @@ exports[`Select renders only the Textbox when closed 1`] = `
 </styled.div>
 `;
```
Comment (mkrds) [inline src/__experimental__/components/select/__snapshots__/select.spec.js.snap]: I think you forgot to remove this :)

Comment (damianOfKyan) [reply src/__experimental__/components/select/__snapshots__/select.spec.js.snap]: yep, fixed :)

**Pull request merged**